Framework/CurveFitting/src/Functions/TabulatedFunction.cpp

```diff
@@ -275,6 +275,9 @@ void TabulatedFunction::load(const std::string &fname) {
 void TabulatedFunction::loadWorkspace(const std::string &wsName) const {
   auto ws = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(wsName);
   loadWorkspace(ws);
+  if (!m_workspace) {
+    throw std::runtime_error("Unable to set " + wsName + " as workspace attribute. Incorrect workspace type?");
```

Comment:
```suggestion
    throw std::runtime_error("Unable to set " + wsName + " as workspace attribute. Expected a MatrixWorkspace.");
```
Could we have the error message inform the user of what workspaces are allowed?